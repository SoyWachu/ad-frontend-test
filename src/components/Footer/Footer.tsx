import Image from "next/image";
import Link from "next/link";

export default function Footer() {
  return (
    <footer className="flex justify-center items-center py-16 text-white bg-footer">
      <Link href="/">
        <Image
          src="/logos/apply-digital-logo.png"
          alt="Logo Apply Digital"
          className="h-10"
          width={170}
          height={44}
        />
      </Link>
    </footer>
  );
}
