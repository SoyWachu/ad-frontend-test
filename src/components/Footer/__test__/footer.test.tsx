import { render, screen } from "@testing-library/react";
import Footer from "../Footer";

jest.mock("next/image", () => ({
  __esModule: true,
  default: (props: any) => <img alt={props.alt || ""} {...props} />,
}));

describe("Footer", () => {
  it("renders logo correctly", () => {
    render(<Footer />);
    const logo = screen.getByAltText("Logo Apply Digital");
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("src", "/logos/apply-digital-logo.png");
  });

  it("matches snapshot", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
