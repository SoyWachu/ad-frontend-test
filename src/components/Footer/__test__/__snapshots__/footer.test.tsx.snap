// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Footer matches snapshot 1`] = `
<div>
  <footer
    class="flex justify-center items-center py-16 text-white bg-footer"
  >
    <a
      href="/"
    >
      <img
        alt="Logo Apply Digital"
        class="h-10"
        height="44"
        src="/logos/apply-digital-logo.png"
        width="170"
      />
    </a>
  </footer>
</div>
`;
